### DISCUSSION TITLE
Q&A area for implementing DB2 support in z390

### DISCUSSION THREAD

**Discussion by abekornelis** [z390development/z390#464] [Q&A]
## DB2 support in z390
  We’re using this discussion as a place to share and exchange info related to implementing DB2 support in z390
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.

For the maintainers, here are some tips 💡 for getting started with Discussions.

  📢 **Announce to your community** that Discussions is available! Go ahead and send that tweet, post, or link it from the website to drive traffic here.

  🔗 If you use issue templates, **link any relevant issue templates** such as questions and community conversations to Discussions. Declutter your issues by driving community content to where they belong in Discussions. If you need help, here's a [link to the documentation](https://docs.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository#configuring-the-template-chooser).

  ➡️ You can **convert issues to discussions** either individually or bulk by labels. Looking at you, issues labeled “question” or “discussion”.

**Comment by CPSstudent** [comment]


**Reply by adelosa** [reply]
This is probably not the right spot for requesting help but I'd be happy to assist you getting started with z390. Please raise via a new issue [here](https://github.com/z390development/z390/issues) and we work through the issue there.

**Comment by chookperson** [comment]
Abe,

I followed your instructions and following is the "command prompt" session:

C:\Program Files\IBM\SQLLIB\BIN>cd c:\

c:\>md z390dev

c:\>cd z390dev

c:\z390dev>git clone https://github.com/chookperson/z390.git z390
Cloning into 'z390'...
remote: Enumerating objects: 10052, done.
remote: Total 10052 (delta 0), reused 0 (delta 0), pack-reused 10052
Receiving objects: 100% (10052/10052), 8.15 MiB | 4.75 MiB/s, done.
Resolving deltas: 100% (7247/7247), done.
Updating files: 100% (2625/2625), done.

c:\z390dev>cd z390

c:\z390dev\z390> git status
On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean

c:\z390dev\z390>git branch -v --all
* main                0f9cdbd Fix X2C issue in DFHMDF macro (#457)
  remotes/origin/HEAD -> origin/main
  remotes/origin/main 0f9cdbd Fix X2C issue in DFHMDF macro (#457)

c:\z390dev\z390>

John Hennesy
p.s. Thanks for your patience

**Comment by abekornelis** [comment]
John, you're welcome! And thank you for your persistence!
Results are exactly what we wanted to achieve - well done!

The next step would be to create your own development branch and switch into it.

As before, please open a command prompt and cd into the main dir of your clone.
C:
CD \z390dev\z390

Then issue this command:
git switch -c DB2
git branch -v --all

The last command should show your new branch  in green, as your current branch.
Please copy/paste all commands and output again in here.
I'm sure you'll be all right - but I'm a control freak :-)

Success again!
Abe

**Comment by abekornelis** [comment]
John,

once you have established your branch, you can start validation of the environment.

After opening a DOS box and cd-ing into c:\z390dev\z390 you'll want to try and create the z390 executable, and run the base set of regression tests. You can do this by issuing
build.bat

This script will compile z390 into a jar, then build all other z390 components and run a whole series of regression tests.

Please let me know how this works our on your system.
Abe
